packages/http/src/models/Role.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable max-classes-per-file */
+import DataModel from './DataModel'
+
+export interface RoleJsonObject {
+  type: string
+  time: number
+  role: string
+  height: number
+  hash: string
+}
+
+export class RoleObject extends DataModel {
+  type!: string
+
+  time!: number
+
+  role!: string
+
+  height!: number
+
+  hash!: string
+
+  constructor(data: Role) {
+    super()
+    Object.assign(this, data)
+  }
+
+  get data(): Role {
+    return this
+  }
+}
+
+export type AnyRole = RoleObject
+
+export default class Role {
+  public static fromJsonObject(json: RoleJsonObject): AnyRole {
```

Comment:
not sure you need these either. Role can just be built from a constructor. txns are trickier because we're turning lots of different types of input into the object